include/oneapi/dpl/pstl/hetero/numeric_impl_hetero.h

```diff
@@ -131,21 +131,23 @@ __pattern_transform_scan_base_impl(_ExecutionPolicy&& __exec, _Iterator1 __first
         if constexpr (__can_use_group_scan)
         {
             oneapi::dpl::__par_backend_hetero::__pattern_transform_scan_single_group(
-                std::forward<_ExecutionPolicy>(__exec), __buf1, __buf2, __n, __unary_op, __init, __binary_op,
-                _Inclusive{})
+                ::std::forward<_ExecutionPolicy>(__exec), __buf1.all_view(), __buf2.all_view(), __n, __unary_op, __init,
+                __binary_op, _Inclusive{})
                 .wait();
         }
         else
         {
             oneapi::dpl::__par_backend_hetero::__pattern_transform_scan_multi_group(
-                std::forward<_ExecutionPolicy>(__exec), __buf1, __buf2, __unary_op, __init, __binary_op, _Inclusive{})
+                ::std::forward<_ExecutionPolicy>(__exec), __buf1.all_view(), __buf2.all_view(), __unary_op, __init,
+                __binary_op, _Inclusive{})
                 .wait();
         }
     }
     else
     {
         oneapi::dpl::__par_backend_hetero::__pattern_transform_scan_multi_group(
-            std::forward<_ExecutionPolicy>(__exec), __buf1, __buf2, __unary_op, __init, __binary_op, _Inclusive{})
+            ::std::forward<_ExecutionPolicy>(__exec), __buf1.all_view(), __buf2.all_view(), __unary_op, __init,
+            __binary_op, _Inclusive{})
             .wait();
     }
```

Comment:
As far as I can see, there is no difference between lines 140-143 and 148-151. Can the conditions be reorganized to avoid duplication? For example, like in this pseudocode:
```
if constexpr (can use group scan) {
    if (runtime conditions to use a single group) {
        use a single group;
        return;
    }
}
use all groups;
```

Reply:
Thanks! I couldn't figure out how to combine the compile-time and runtime conditions in a way that avoids code duplication. But using an early return as you suggest makes this a lot cleaner.